CMakeLists.txt

```diff
@@ -88,44 +87,42 @@ list(PREPEND CMAKE_PREFIX_PATH ${CMAKE_CURRENT_BINARY_DIR}/faabric)
 include(${CMAKE_CURRENT_BINARY_DIR}/faabric/conan_paths.cmake)
 find_package(cpprestsdk REQUIRED)
 
-# Detect SGX
+# ----------------------------------------
+# SGX configuration
+# ----------------------------------------
+
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
-if (FAASM_SGX)
+if (NOT (FAASM_SGX_MODE STREQUAL "Disabled"))
+    # Detect SGX intallation
     find_package(SGX)
-endif ()
 
-# SGX configuration
-if (SGX_FOUND)
-    # SGX includes
-    include_directories(${SGX_SDK_PATH}/include)
-
-    # Set up SGX definitions
-    add_definitions(-DFAASM_SGX=1)
-
-    if(FAASM_SGX_SIM_MODE)
-        add_definitions(-DSGX_SIM_MODE=1)
-        set(Protobuf_USE_STATIC_LIBS 0)
-    else()
-	    add_definitions(-DSGX_SIM_MODE=0)
-        # Must link protobuf statically because SGX LE probably uses an older version
-        set(Protobuf_USE_STATIC_LIBS 1)
-    endif()
-
-    if(FAASM_SGX_ATTESTATION)
-        add_definitions(-DFAASM_SGX_ATTESTATION=1)
-    else()
-        add_definitions(-DFAASM_SGX_ATTESTATION=0)
-    endif()
+    # SGX configuration
```

Comment:
We only try to find the SGX installation if SGX is not disabled from the config.

If we look for it, and don't find it, we must error out.

Note that this could only possibly happen in out-of-container builds.